pkg/grid/src/preferences/SearchSystemPrefences.tsx

```diff
@@ -0,0 +1,154 @@
+import LogoutIcon from '../components/icons/LogoutIcon';
+import PencilIcon from '../components/icons/PencilIcon';
+import ForwardSlashIcon from '../components/icons/ForwardSlashIcon';
+
+const navOptions: { route: string; title: string; icon: React.ReactElement }[] = [
+  {
+    route: 'help',
+    title: 'Help and Support',
+    icon: <HelpIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'interface',
+    title: 'Interface Settings',
+    icon: <Interface className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'notifications',
+    title: 'Notifications',
+    icon: <BellIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'appearance',
+    title: 'Appearance',
+    icon: <PencilIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'shortcuts',
+    title: 'Shortcuts',
+    icon: <ForwardSlashIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'privacy',
+    title: 'Attention & Privacy',
+    icon: <BurstIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'security',
+    title: 'Log Out...',
+    icon: <LogoutIcon className="w-4 h-4 text-gray-600" />
+  },
+  {
+    route: 'system-updates',
+    title: 'About System',
+    icon: <TlonIcon className="w-4 h-4 text-gray-600" />
+  }
+];
+
+interface SearchSystemPrefencesProps {
+  subUrl: (submenu: string) => string;
+}
+
+export default function SearchSystemPreferences({ subUrl }: SearchSystemPrefencesProps) {
+  const { push } = useHistory();
+  const [searchInput, setSearchInput] = useState('');
+  const [matchingNavOptions, setMatchingNavOptions] = useState<string[]>([]);
+  const [highlightNavOption, setHighlightNavOption] = useState<number>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const input = e.target as HTMLInputElement;
+    const value = input.value.trim();
+
+    setSearchInput(value);
+  };
+
+  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    const { key } = e;
+    if (key === 'ArrowDown' && searchInput !== '' && matchingNavOptions.length > 0) {
+      if (highlightNavOption === undefined) {
+        setHighlightNavOption(0);
+      } else {
+        setHighlightNavOption((prevState) => prevState! + 1);
+      }
+    }
+
+    if (
+      key === 'ArrowUp' &&
+      searchInput !== '' &&
+      matchingNavOptions.length > 0 &&
+      highlightNavOption !== undefined &&
+      highlightNavOption !== 0
+    ) {
+      setHighlightNavOption((prevState) => prevState! - 1);
+    }
+
+    if (key === 'Enter' && searchInput !== '' && highlightNavOption !== undefined) {
+      push(subUrl(navOptions[highlightNavOption].route));
+    }
+  };
+
+  const handleBlur = () => {
+    setSearchInput('');
+  };
+
+  useEffect(() => {
+    const results = fuzzy.filter(searchInput, navOptions, { extract: (obj) => obj.title });
+    const matches = results.map((el) => el.string);
+    setMatchingNavOptions(matches);
+  }, [searchInput]);
```

Comment:
should this be thrown into a debounce or similar?

Reply:
I don't think it's necessary since we're not hitting the backend.